### PR TITLE
Feature/user class user stories

### LINKED ISSUE #6: should be able to view a list of all recipes
As a user, I should be able to view a list of all recipes.

### LINKED ISSUE #13: User can favorite/ unfavorite recipes
As a user, I should be able to favorite / unfavorite recipes that I like and can easily find again.

### LINKED ISSUE #14: User can filter by favorite and tags
As a user, I should be able to filter my favorited recipes by one or more tags.

### LINKED ISSUE #15: User can sort by favorite and info
As a user, I should be able to search my favorited recipes by its name or ingredients.

### LINKED ISSUE #16: User can see/ access recipes to cook
As a user, I should be able to add a recipe to a list of recipes to cook.

### PR DESCRIPTION
**Pull request by Logandv3**
## Description

Please include a summary of the change and/or functionality implementation:
This pull request connects the User class methods to the script file and allows the user to search through favorite recipes by name, ingredient or tags.  It also allows the user to add recipes to a cooking list.

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] open index.html
- [x] dev tools

## Checklist:

- [x] I have performed a self-review of my own code
- [x] No console error messages

## Additional info
Please include any other relevant information here:
New issues have been added to the project board.

closes #6
closes #13 
closes #14 
closes #15 
closes #16

### CODE REVIEW

File: src/classes/User.js [hunk 1]
```diff
@@ -33,63 +33,72 @@ class User {
   removeFromRecipesToCook(recipeOut) {
     this.recipesToCook.forEach(recipe => {
       if (recipeOut.name === recipe.name) {
-        this.recipesToCook.splice(recipeOut, 1)
+        this.recipesToCook.splice(this.recipesToCook.indexOf(recipeOut), 1)
       }
     })
   }
 
-  filterFavoriteRecipeTags(tags) {
```
Comment (elisebeall) [inline src/classes/User.js:41]: I watched you type all of these changes.  They look good.  And they work!

File: src/classes/User.js [hunk 2]
```diff
@@ -33,63 +33,72 @@ class User {
...
-      return "No recipes match criteria."
-    } else {
-      return matchTags
-    }
+  filterFavoriteRecipeTags(inputTag) {
+    let tagged = this.favoriteRecipes.filter(recipe => recipe.tags.includes(inputTag));
+    return tagged;
   }
 
-  filterFavoriteRecipeByName(name) {
```
Comment (elisebeall) [inline src/classes/User.js:57]: These are simple adjustments to use the same term for the search input, across all classes

**Review by EricSergeant — APPROVED** [review]
Good solution of updating buttons for search when within all recipes, favorites, and to cook.  Also, great problem-solving to get all functionality running to this point!

File: src/classes/User.js [hunk 3]
```diff
@@ -33,63 +33,72 @@ class User {
...
+  filterFavoriteRecipeByName(searchInput) {
     let matchName = [];
     this.favoriteRecipes.forEach(recipe => {
-      if (recipe.name.includes(name)) {
+      if (recipe.name.toLowerCase().includes(searchInput.toLowerCase())) {
         matchName.push(recipe);
+        console.log('recipe', recipe);
       }
     })
-    if (!matchName.length) {
```
Comment (elisebeall) [inline src/classes/User.js:64]: This was causing some strange behavior when the conditional was matched.  It is also unnecessary if we implement warnings for if the user's search returns zero results.

File: src/classes/User.js [hunk 4]
```diff
@@ -33,63 +33,72 @@ class User {
...
-        if (ingredient.id === matchedName.id) {
-          return matchIngred.push(recipe)
-        }
+        matchedName.forEach(id => {
+          if (ingredient.id === id.id) {
+            return matchIngred.push(recipe)
+          }})
+        })
       })
-    })
```
Comment (elisebeall) [inline src/classes/User.js:85]: Same comment as above!

File: src/index.html [hunk 5]
```diff
@@ -10,19 +10,21 @@
...
-      <button type="button" name="User Name" id="userName">User Name</button>
+      <h2 id="userName"></h2>
+      <button type="button" name="All Recipes" id="allRecipesBtn">All Recipes</button>
       <button type="button" name="Favorite Recipes" id="favoriteRecipes">Favorite Recipes</button>
       <button type="button" name="What to Cook" id="whatToCook">What to Cook</button>
     </nav>
     <main>
       <section class="search-box">
         <form class="search-options">
-          <input type="text" name="search" placeholder="Search Recipe by Name or Indgredient" id="searchBar">
```
Comment (elisebeall) [inline src/index.html:20]: Went with the easiest solution that we could think of!  It involved creating new search buttons to hide and show on the DOM based on what was being searched (all recipes, favorite recipes).

File: src/index.html [hunk 6]
```diff
@@ -36,10 +38,10 @@ <h1>What's Cookin'</h1>
         <button type="button" name="back-to-main" class="back-to-main-btn" id="backToMainBtn">Back to Main</button>
         <p id="recipeTitle">Elvis Pancakes</p>
         <img id="indRecipeImage" src="https://spoonacular.com/recipeImages/741603-556x370.jpeg" alt="Elvis Pancakes">
-        <h3 id="addToFavoriteList">☆ Add to Favorite List</h3>
-        <h3 id="onFavoriteList">★ On Favorite List</h3>
-        <h3 id="addToCookingList">☐ Add to Cooking List</h3>
-        <h3 id="onCookingList">☑︎ On Cooking List</h3>
+        <h3 class="list-btns" id="addToFavoriteList">☆ Add to Favorite List</h3>
```
Comment (elisebeall) [inline src/index.html:41]: Hiding and showing buttons based on whether or not the recipe is contained in the array.

**Pull request merged**